src/core/config/type-cast/type-casters/number.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * @prettier
+ */
+const numberTypeCaster = (value) => {
+  const parsedValue = parseInt(value, 10)
+  return Number.isNaN(parsedValue) ? NaN : parsedValue
```

Comment:
I was thinking of returning default values here instead of `NaN`, but then we would need to go back to checking the type of `maxDisplayedTags`

Reply:
Choose the new default: -1 or NaN is just fine, but -1 looks better